src/javascript/app/pages/trade/digit_ticker.js

```diff
@@ -0,0 +1,101 @@
+const DIGIT_BLOCK_SIZE        = 36;
+const DIGIT_BLOCK_LEFT_OFFSET = 66;
+const DIGIT_BLOCK_CORRECTION  = 2;
+
+const DigitTicker = (() => {
+    let $container, source;
+    const digits = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+    const init   = (container_id, contract) => {
+        source = contract;
+
+        $container = $(`#${container_id}`);
+        $container
+            .append($('<div />', { class: 'peek-box' })
+                .append($('<div />', { class: 'mask', text: '0/0' }))
+                .append($('<div />', { class: 'peek' })))
+            .append($('<div />', {
+                class: 'digits', html : digits.map(digit => `<div class='digit digit-${digit}'>${digit}</div>`).join(''),
+            }));
+        colorWinningBoxes(winningNumbers(contract));
+    };
+
+    const winningNumbers = (contract) => {
+        switch (contract.contract_type) {
+            case 'DIGITOVER':
+                return digits.filter(digit => +digit > +contract.barrier);
+            case 'DIGITUNDER':
+                return digits.filter(digit => +digit < +contract.barrier);
+            case 'DIGITMATCH':
+                return digits.filter(digit => +digit === +contract.barrier);
+            case 'DIGITDIFF':
+                return digits.filter(digit => +digit !== +contract.barrier);
+            case 'DIGITODD':
+                return digits.filter(digit => +digit % 2 !== 0);
+            case 'DIGITEVEN':
+                return digits.filter(digit => +digit % 2 === 0);
+            default:
+                throw new Error('Cannot Determine Winning numbers.');
+        }
+    };
+
+    const colorWinningBoxes = (winning_numbers) => {
+        winning_numbers.forEach(digit => {
+            $(`.digit-${digit}`).css({
+                backgroundColor: '#9ddfcb',
+                color          : '#67b9a0',
+            });
```

Comment:
it's better to add a digit-winning or similar class to it instead, and keep the styling for that class in the css file. we don't want css styling in javascript files.